src/dune_rules/cram_exec.ml

```diff
@@ -135,6 +135,13 @@ let cram_stanzas lexbuf =
 
 let run_expect_test file ~f =
   let file_contents = Io.read_file ~binary:false file in
+  (* nasty hack so that the user doesn't observe the test file they are running.
+
+     Eventually, we should just have a way to read the source from outside the
+     sandbox. *)
+  let temp_dir = Temp.create Dir ~prefix:"cram-staging" ~suffix:"" in
```

Comment:
Shouldn't we delete `temp_dir` eventually?

BTW, we already have the whole file contents in memory (in the variable `file_contents`), so instead of moving it we could simply delete it and re-create it with `Io.write_file` in case of mismatch.

Reply:
The temp_dir is deleted on at_exit. I still changed to it to use write_file

Reply:
We should still delete temporary things when we know they are no longer needed, to avoid holding on to leftover temporary files and directories for too long. This is especially important when using the watch mode.